app/app.go

```diff
@@ -414,6 +415,27 @@ func New(
 	)
 	app.SetEndBlocker(app.EndBlocker)
 
+	planName := "v2.0.0"
+	app.UpgradeKeeper.SetUpgradeHandler(planName, func(ctx sdk.Context, _ upgradetypes.Plan) {
+		// https://github.com/crypto-org-chain/chain-main/blob/master/doc/architecture/adr-003.md
+	})
+
+	upgradeInfo, err := app.UpgradeKeeper.ReadUpgradeInfoFromDisk()
+	if err != nil {
+		panic(err)
+	}
+
+	if upgradeInfo.Name == planName && !app.UpgradeKeeper.IsSkipHeight(upgradeInfo.Height) {
+		storeUpgrades := storetypes.StoreUpgrades{
+			Added:   []string{nfttypes.StoreKey},
+			Renamed: []storetypes.StoreRename{},
+			Deleted: []string{},
+		}
+
+		// configure store loader that checks if version == upgradeHeight and applies store upgrades
+		app.SetStoreLoader(upgradetypes.UpgradeStoreLoader(upgradeInfo.Height, &storeUpgrades))
```

Comment:
should this just be called once / inside `SetUpgradeHandler` ?

Reply:
I think the override only works before storage constructed, so it need to happen pretty early. while upgrade handler is called in abci event.

Reply:
I get the example code from the godoc of upgrade module: https://pkg.go.dev/github.com/cosmos/cosmos-sdk@v0.42.4/x/upgrade

Reply:
https://github.com/cosmos/cosmos-sdk/blob/master/x/upgrade/types/storeloader.go#L13
And before apply the upgrade, it'll check the height again, so it should be safe here.